worker/worker.py

```diff
@@ -95,31 +95,14 @@
 
 bool_list = ["true", "1", "false", "0"]
 
-# A custom boolean type that can be instantiated
-# from a string.
-class BOOL(object):
-    def __init__(self, x):
-        y = x.lower().strip()
-        if y not in bool_list:
-            raise ValueError(x)
-        self.x = y
-
-    def __bool__(self):
-        if self.x in ["0", "1"]:
-            return self.x == "1"
-        else:
-            return self.x.lower() == "true"
-
-    def __eq__(self, w):
-        if not isinstance(w, BOOL) and not isinstance(w, bool):
-            return False
-        return self.__bool__() == w.__bool__()
-
-    def __str__(self):
-        return str(self.__bool__())
-
-    __repr__ = __str__
-
+def BOOL(x):
```

Comment:
I saw that too but I preferred not to depend on an external function. That way can tweak the recognition function more easily to our needs (it's very small anyway).

Reply:
Ok. Notes:
- leftover ? `BOOL.__name__ = "bool"`
- here are suggesting to use `str2bool()` to avoid confusion with the meaning of standard `bool()` https://stackoverflow.com/questions/15008758/parsing-boolean-values-with-argparse

```python
def str2bool(x):
    x = x.strip().lower()
    if x in {"true", "1"}:
        return True
    elif x in {"false", "0"}:
        return False
    else:
        raise ValueError(x)
```

BTW `distutils.util.strtobool()` is broken (`return 1` is not `return True`). Fortunately `distutil` is deprecated and will be dropped in python 3.12
https://github.com/python/cpython/blob/98fac8bc183c113903403793ffe411358ee40d8a/Lib/distutils/util.py#L308-L321

```python
import distutils.util
print(f"{distutils.util.strtobool('True') is True}")
```